aws/resource_aws_globalaccelerator_endpoint_group.go

```diff
@@ -0,0 +1,360 @@
+	}
+
+	if v, ok := d.GetOk("health_check_protocol"); ok {
+		opts.HealthCheckProtocol = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("threshold_count"); ok {
+		opts.ThresholdCount = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("traffic_dial_percentage"); ok {
+		opts.TrafficDialPercentage = aws.Float64(v.(float64))
+	}
+
+	if v, ok := d.GetOk("endpoint_configurations"); ok {
+		opts.EndpointConfigurations = resourceAwsGlobalAcceleratorEndpointGroupExpandEndpointConfigurations(v.([]interface{}))
+	}
+
+	log.Printf("[DEBUG] Create Global Accelerator endpoint group: %s", opts)
+
+	resp, err := conn.CreateEndpointGroup(opts)
+	if err != nil {
+		return fmt.Errorf("Error creating Global Accelerator endpoint group: %s", err)
+	}
+
+	d.SetId(*resp.EndpointGroup.EndpointGroupArn)
+
+	acceleratorArn, err := resourceAwsGlobalAcceleratorListenerParseAcceleratorArn(d.Id())
+	if err != nil {
+		return err
+	}
+
+	// Creating an endpoint group triggers the accelerator to change status to InPending
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{globalaccelerator.AcceleratorStatusInProgress},
+		Target:  []string{globalaccelerator.AcceleratorStatusDeployed},
+		Refresh: resourceAwsGlobalAcceleratorAcceleratorStateRefreshFunc(conn, acceleratorArn),
+		Timeout: d.Timeout(schema.TimeoutCreate),
+	}
+
+	log.Printf("[DEBUG] Waiting for Global Accelerator endpoint group (%s) availability", d.Id())
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error waiting for Global Accelerator endpoint group (%s) availability: %s", d.Id(), err)
+	}
+
+	return resourceAwsGlobalAcceleratorEndpointGroupRead(d, meta)
+}
+
+func resourceAwsGlobalAcceleratorEndpointGroupRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).globalacceleratorconn
+
+	endpointGroup, err := resourceAwsGlobalAcceleratorEndpointGroupRetrieve(conn, d.Id())
+
+	if err != nil {
+		return fmt.Errorf("Error reading Global Accelerator endpoint group: %s", err)
+	}
+
+	if endpointGroup == nil {
+		log.Printf("[WARN] Global Accelerator endpoint group (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	listenerArn, err := resourceAwsGlobalAcceleratorEndpointGroupParseListenerArn(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("listener_arn", listenerArn)
+	d.Set("endpoint_group_region", endpointGroup.EndpointGroupRegion)
+	d.Set("health_check_interval_seconds", endpointGroup.HealthCheckIntervalSeconds)
+	d.Set("health_check_path", endpointGroup.HealthCheckPath)
+	d.Set("health_check_port", endpointGroup.HealthCheckPort)
+	d.Set("health_check_protocol", endpointGroup.HealthCheckProtocol)
+	d.Set("threshold_count", endpointGroup.ThresholdCount)
+	d.Set("traffic_dial_percentage", endpointGroup.TrafficDialPercentage)
+	d.Set("endpoint_configurations", resourceAwsGlobalAcceleratorEndpointGroupFlattenEndpointDescriptions(endpointGroup.EndpointDescriptions))
```

Comment:
When using `d.Set()` with aggregate types (TypeList, TypeSet, TypeMap), [we should perform error checking to prevent issues where the code is not properly able to set the Terraform state](https://www.terraform.io/docs/extend/best-practices/detecting-drift.html#error-checking-aggregate-types). e.g.

```suggestion
	if err := d.Set("endpoint_configurations", resourceAwsGlobalAcceleratorEndpointGroupFlattenEndpointDescriptions(endpointGroup.EndpointDescriptions)); err != nil {
		return fmt.Errorf("error setting endpoint_configurations: %s", err)
	}
```